### PR TITLE
Add overlay

### PR DESCRIPTION
**Pull request by thompsond**
- Add image overlay
- Add batch download

### CODE REVIEW

File: src/main/webapp/js/script.js [hunk 1]
```diff
@@ -26,23 +26,83 @@ $(document).ready(function() {
...
                 atag.appendChild(image);
+                atag.appendChild(h4);
                 div.appendChild(atag);
-                div.appendChild(h4);
                 $('#requested-images').append(div);
             }
+            if (array.length > 0) {
+                $('#download-btn').removeClass('disabled');
+            }
+            else {
```
Comment (gabbygermanson) [inline src/main/webapp/js/script.js:44]: combine else with if end bracket

File: src/main/webapp/js/script.js [hunk 2]
```diff
@@ -26,23 +26,83 @@ $(document).ready(function() {
...
+
+    $('#requested-images').on('click', '.tile a', function(event) {
+        let gridImage = $(this).children('img');
+        let url = gridImage.attr('src');
+        $('#overlay img').attr('src', url);
+        $('#overlay').show();
+    });
+
+    $("#overlay").click(function(event) {
+        if(event.target.id !== "overlay-img") {
```
Comment (gabbygermanson) [inline src/main/webapp/js/script.js:58]: space after if

File: src/main/webapp/js/script.js [hunk 3]
```diff
@@ -26,23 +26,83 @@ $(document).ready(function() {
...
     });
 
     // Upload files any files to Drive that need to be uploaded
     fetch('/start-save-drive');
 });
 
+function urlToPromise(url) {
+    return new Promise(function(resolve, reject) {
+        JSZipUtils.getBinaryContent(url, function (err, data) {
+            if(err) {
```
Comment (gabbygermanson) [inline src/main/webapp/js/script.js:97]: space after if

File: src/main/java/com/google/step/FrontendQueryDatastore.java [hunk 4]
```diff
@@ -70,7 +66,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
                 buildCompositeFilter(zoomStrings, cityStrings, startDateStr, endDateStr);
 
         // Build the query for Datastore.
-        Query query = new Query("MapImage").setFilter(compositeFilter);
+        Query query =
+                new Query("MapImage")
+                        .setFilter(compositeFilter)
+                        .addSort("City Name", SortDirection.ASCENDING)
+                        .addSort("Zoom", SortDirection.ASCENDING)
+                        .addSort("Month", SortDirection.ASCENDING);
```
Comment (cgregori) [inline src/main/java/com/google/step/FrontendQueryDatastore.java:74]: Did this sort (Month) work? I was having some issues earlier in the month with sorting by month and I'm not sure if the changes to index.yaml have fixed them

Comment (thompsond) [reply src/main/java/com/google/step/FrontendQueryDatastore.java:74]: Yea it's something like Canadian 7/2020 5x, Canadian 13/2020 5x, Delhi 7/2020 5x

**Pull request merged**